src/heretic/model.py

```diff
@@ -84,12 +87,13 @@ def __init__(self, settings: Settings):
 
     def reload_model(self):
         dtype = self.model.dtype
+        model_class = self.model.__class__
 
         # Purge existing model object from memory to make space.
         self.model = None
         empty_cache()
 
-        self.model = AutoModelForCausalLM.from_pretrained(
+        self.model = model_class.from_pretrained(
```

Comment:
There are **STILL** changes unrelated to notebooks in this PR! This is the fourth time I'm pointing that out. Please check the diff before requesting a review again. There should be **NO** changes to `model.py`.